### PR TITLE
Resolve conflicts in toplevel/

### PR DESCRIPTION
**Pull request by poechsel**
Most of this files depends on `dynlink_platform_intf.ml` which has not been fixed yet. So to test it I:
1. Modified `dynlink_platform_intf.ml` just enough so that the typechecker wouldn't complain
2. Executed
```
dune b --workspace=duneconf/boot.ws \
	_build/default/toplevel/byte/.ocamltoplevel.objs/byte/genprintval.cmt \
	_build/default/toplevel/byte/.ocamltoplevel.objs/byte/genprintval.cmti \
	_build/default/toplevel/byte/.ocamltoplevel.objs/byte/topcommon.cmt \
	_build/default/toplevel/byte/.ocamltoplevel.objs/byte/topcommon.cmti \
	_build/default/toplevel/byte/.ocamltoplevel.objs/byte/topdirs.cmt \
	_build/default/toplevel/byte/.ocamltoplevel.objs/byte/topdirs.cmti \
	_build/default/toplevel/byte/.ocamltoplevel.objs/byte/topeval.cmt \
	_build/default/toplevel/byte/.ocamltoplevel.objs/byte/topeval.cmti \
	_build/default/toplevel/byte/.ocamltoplevel.objs/byte/toploop.cmt \
	_build/default/toplevel/byte/.ocamltoplevel.objs/byte/toploop.cmti \
	_build/default/toplevel/byte/.ocamltoplevel.objs/byte/topmain.cmt \
	_build/default/toplevel/byte/.ocamltoplevel.objs/byte/topmain.cmti \
	_build/default/toplevel/byte/.ocamltoplevel.objs/byte/topprinters.cmt \
	_build/default/toplevel/byte/.ocamltoplevel.objs/byte/topprinters.cmti \
	_build/default/toplevel/byte/.ocamltoplevel.objs/byte/trace.cmt \
	_build/default/toplevel/byte/.ocamltoplevel.objs/byte/trace.cmti \
	_build/default/toplevel/native/.ocamltoplevel_native.objs/byte/genprintval.cmt \
	_build/default/toplevel/native/.ocamltoplevel_native.objs/byte/genprintval.cmti \
	_build/default/toplevel/native/.ocamltoplevel_native.objs/byte/topcommon.cmt \
	_build/default/toplevel/native/.ocamltoplevel_native.objs/byte/topcommon.cmti \
	_build/default/toplevel/native/.ocamltoplevel_native.objs/byte/topdirs.cmt \
	_build/default/toplevel/native/.ocamltoplevel_native.objs/byte/topdirs.cmti \
	_build/default/toplevel/native/.ocamltoplevel_native.objs/byte/topeval.cmt \
	_build/default/toplevel/native/.ocamltoplevel_native.objs/byte/topeval.cmti \
	_build/default/toplevel/native/.ocamltoplevel_native.objs/byte/tophooks.cmt \
	_build/default/toplevel/native/.ocamltoplevel_native.objs/byte/tophooks.cmti \
	_build/default/toplevel/native/.ocamltoplevel_native.objs/byte/toploop.cmt \
	_build/default/toplevel/native/.ocamltoplevel_native.objs/byte/toploop.cmti \
	_build/default/toplevel/native/.ocamltoplevel_native.objs/byte/topmain.cmt \
	_build/default/toplevel/native/.ocamltoplevel_native.objs/byte/topmain.cmti \
	_build/default/toplevel/native/.ocamltoplevel_native.objs/byte/topprinters.cmt \
	_build/default/toplevel/native/.ocamltoplevel_native.objs/byte/topprinters.cmti \
	_build/default/toplevel/native/.ocamltoplevel_native.objs/byte/trace.cmt \
	_build/default/toplevel/native/.ocamltoplevel_native.objs/byte/trace.cmti \
	_build/default/toplevel/native/.topstart.eobjs/byte/topstart.cmt \
	_build/default/toplevel/native/.topstart.eobjs/byte/topstart.cmti \
	_build/default/toplevel/.topstart.eobjs/byte/topstart.cmt \
	_build/default/toplevel/.topstart.eobjs/byte/topstart.cmti 
```

### CODE REVIEW

**Comment by ncik-roberts** [comment]
Reviewing

**Comment by ncik-roberts** [comment]
I'm pulling in the same change to `dynlink_platform_intf.ml` as I did in https://github.com/ocaml-flambda/ocaml-jst/pull/199

File: toplevel/topdirs.ml [hunk 1]
```diff
@@ -728,53 +622,25 @@ let () =
 let () =
   reg_show_prim "show_class"
     (fun env loc id lid ->
-<<<<<<< HEAD
-       let path, desc_class = Env.lookup_class ~loc lid env in
-       let _path, desc_cltype = Env.lookup_cltype ~loc lid env in
-       let _path, typedcl = Env.lookup_type ~loc lid env in
-       let hash_typedcl = Env.find_hash_type path env in
```
Comment (ncik-roberts) [inline toplevel/topdirs.ml:735]: Intentional for this to be dropped? It seems reasonable to keep in to me, but I might be missing context.

**Review by ncik-roberts — APPROVED** [review]
Looks good to me. Just one minor question.

File: toplevel/topdirs.ml (hunk 1 above)
Comment (poechsel) [reply toplevel/topdirs.ml:735]: It is not dropped - only the `hash_typedcl` bit is dropped
This is due to https://github.com/ocaml/ocaml/commit/1632e9b2cfaee20d9921b9e2828abb7931a376ad

**Pull request merged**